runtime/common/Cargo.toml

```diff
@@ -40,6 +40,11 @@ pallet-evm = { git = "https://github.com/PureStake/frontier", default-features =
 pallet-evm-chain-id = { git = "https://github.com/PureStake/frontier", default-features = false, branch = "moonbeam-polkadot-v0.9.38" }
 pallet-evm-precompile-dispatch = { git = "https://github.com/PureStake/frontier", default-features = false, branch = "moonbeam-polkadot-v0.9.38" }
 
+# There is a bug in the frontier repo that adds pallet-ethereum without a try-runtime dependency
+# for this crate which makes our compilation fail with the i_know_what_i_am_doing error.
+# It seem fixed in 0.9.39, and this dependency can be removed from this file safely.
+fp-self-contained = { git = "https://github.com/PureStake/frontier", default-features = false, branch = "moonbeam-polkadot-v0.9.38" }
+
```

Comment:
I'm pretty sure this was the issue that happened in https://github.com/centrifuge/centrifuge-chain/pull/1406 (not tested yet), so it unlocks that PR.

Reply:
That could very well be it. But I tried using my fork of frontier fixing all the cargo features and the issue still persisted 🤔

Reply:
Have you added `"fp-self-contained/try-runtime"` too to your try?

Reply:
Not sure, but I ran subalfred on the whole repo to make sure I wasn't missing any feature anywhere

Reply:
Ah, but before this PR I also run `subalfred` for all crates, and everything went "well" 😆, and things weren't well. I guess we can not trust 100% in `subalfred`. Some things are not checked.